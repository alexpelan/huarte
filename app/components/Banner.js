import React from 'react';
import {
  Text,
  View,
} from 'react-native';

import {
  dismissError,
} from '../actions/index';

import styles from '../styles/styles';

class Banner extends React.Component {
  dismiss = () => {
    this.props.store.dispatch(dismissError());
  };

  render() {
    return (
      <View style={styles.popove}>
        <Text style={styles.errorPopover}>
          {this.props.errorMessage} &nbsp; &nbsp; &nbsp;
          <Text style={styles.hyperlink} onPress={() => { this.dismiss(); }}>
            Dismiss
          </Text>
        </Text>
      </View>
    );
  }
}

export default Banner;
