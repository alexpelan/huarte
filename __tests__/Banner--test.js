import 'react-native';
import React from 'react';
import Banner from '../app/components/Banner';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <Banner errorMessage={"Something has gone wrong!"} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});