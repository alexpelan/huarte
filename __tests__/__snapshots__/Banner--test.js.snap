// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#ff0033",
        "color": "white",
        "fontSize": 18,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    Something has gone wrong!
          
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "textDecorationLine": "underline",
        }
      }
    >
      Dismiss
    </Text>
  </Text>
</View>
`;
